cmd/ctl/check_config.go

```diff
@@ -0,0 +1,376 @@
+			results := []checkResult{}
+			results = append(results, RunPrincipalChecks(ctx, principalClt, globalOpts.principalNamespace)...)
+			results = append(results, RunAgentChecks(ctx, agentClt.Clientset, globalOpts.agentNamespace, principalClt.Clientset, globalOpts.principalNamespace)...)
+			printResultsAndExit(results)
+		},
+	}
+	return command
+}
+
+func printResultsAndExit(results []checkResult) {
+	hasErr := false
+	fmt.Println("Configuration validation results:")
+	for _, r := range results {
+		fmt.Println(r.String())
+		if r.err != nil {
+			hasErr = true
+		}
+	}
+	if hasErr {
+		cmdutil.Fatal("one or more checks failed")
+	}
+}
+
+// RunPrincipalChecks validates the principal installation and related security assets.
+func RunPrincipalChecks(ctx context.Context, kubeClient *kube.KubernetesClient, principalNS string) []checkResult {
+	out := []checkResult{}
+
+	// Ensure Argo CD is cluster-scoped
+	out = append(out, checkResult{
+		name: "Verifying Argo CD is cluster-scoped",
+		err:  verifyArgoCDClusterScoped(ctx, kubeClient, principalNS),
+	})
+
+	// CA secret exists in principal namespace and is a valid TLS secret
+	out = append(out, checkResult{
+		name: fmt.Sprintf("Verifying principal public CA certificate exists and is valid (%s/%s)", principalNS, config.SecretNamePrincipalCA),
+		err:  principalCheckCA(ctx, kubeClient.Clientset, principalNS),
+	})
+
+	// Principal gRPC TLS secret exists in principal namespace and is valid
+	out = append(out, checkResult{
+		name: fmt.Sprintf("Verifying principal gRPC TLS certificate exists and is valid (%s/%s)", principalNS, config.SecretNamePrincipalTLS),
+		err:  certSecretValid(ctx, kubeClient.Clientset, principalNS, config.SecretNamePrincipalTLS),
+	})
+
+	// Resource proxy TLS secret exists and is valid
+	out = append(out, checkResult{
+		name: fmt.Sprintf("Verifying resource proxy TLS certificate exists and is valid (%s/%s)", principalNS, config.SecretNameProxyTLS),
+		err:  certSecretValid(ctx, kubeClient.Clientset, principalNS, config.SecretNameProxyTLS),
+	})
+
+	// JWT signing key exists
+	out = append(out, checkResult{
+		name: fmt.Sprintf("Verifying JWT signing key exists and is parseable (%s/%s)", principalNS, config.SecretNameJWT),
+		err:  jwtKeyValid(ctx, kubeClient.Clientset, principalNS, config.SecretNameJWT),
+	})
+
+	// Route host matches TLS SANs (OpenShift-only)
+	out = append(out, checkResult{
+		name: "Verifying principal TLS secret ips/dns match Route host (OpenShift)",
+		err:  verifyRouteHostMatchesCert(ctx, kubeClient, principalNS, config.SecretNamePrincipalTLS),
+	})
+
+	return out
+}
+
+// RunAgentChecks validates agent-side security assets and cross-validates them
+// against the principal cluster.
+func RunAgentChecks(ctx context.Context, agentKube kubernetes.Interface, agentNS string, principalKube kubernetes.Interface, principalNS string) []checkResult {
+	out := []checkResult{}
+
+	// Agent CA secret exists and has CA data (opaque secret with ca.crt)
```

Comment:
Similar to my comment in `RunPrincipalChecks`, users can customize the Secret via `.spec.argoCDAgent.agent.(...)` fields, so we should use those values instead, if they are defined.